src/main/java/com/segment/analytics/android/integrations/nielsendcr/NielsenDCRIntegration.java

```diff
@@ -358,6 +368,57 @@ private JSONObject mapSpecialKeys(
     return adContentMetadata;
   }
 
+  public String formatAirdate(String airdate) {
+    String finalDate = airdate;
+
+    // assuming 'airdate' was passed as ISO date string per Segment spec
+    try {
+      Matcher s = shortDate.matcher(finalDate);
+      Matcher l = longDate.matcher(finalDate);
+
+      if (s.find()) {
+        finalDate =
+            new StringBuilder() //
+                .append(s.group(1))
+                .append(s.group(2))
+                .append(s.group(3))
+                .append(" ")
+                .append("00")
+                .append(":")
+                .append("00")
+                .append(":")
+                .append("00")
+                .toString();
+      } else if (l.find()) {
+        finalDate =
+            new StringBuilder() //
+                .append(l.group(1))
+                .append(l.group(2))
+                .append(l.group(3))
+                .append(" ")
+                .append(l.group(4))
+                .append(":")
+                .append(l.group(5))
+                .append(":")
+                .append(l.group(6))
+                .toString();
+      } else {
+        throw new Error("Error parsing airdate from ISO date format.");
+      }
+    } catch (Error e) {
```

Comment:
I'm guessing you would also want to catch unchecked exceptions. In that case do `catch (Throwable t)`